oauth2_provider/models.py

```diff
@@ -401,13 +401,15 @@ def revoke(self):
         access_token_model = get_access_token_model()
         refresh_token_model = get_refresh_token_model()
         with transaction.atomic():
-            self = (
-                refresh_token_model.objects.filter(pk=self.pk, revoked__isnull=True)
-                .select_for_update()
-                .first()
-            )
-            if not self:
+            try:
+                token = refresh_token_model.objects.select_for_update().filter(
+                    pk=self.pk, revoked__isnull=True
+                )
+            except refresh_token_model.DoesNotExist:
```

Comment:
@dag18e @n2ygk I think we added some things here that are not really needed:
1) AFAIK .filter() can't raise DoesNotExist
2) L412 cast the Queryset to a list and then slicing, I do no think this is needed and this slows down things a bit

I feel these things should be addressed in a follow up PR